packages/premint-sdk/src/premint-api.ts

```diff
@@ -0,0 +1,684 @@
+    });
+  }
+
+  /**
+   * Delete premint
+   *
+   * Deletes existing premint
+   * @param settings.collection collection address
+   * @param settings.uid UID
+   * @param settings.walletClient viem wallet client to use to sign
+   *
+   */
+  async deletePremint({
+    walletClient,
+    uid,
+    account,
+    collection,
+  }: {
+    walletClient: WalletClient;
+    uid: number;
+    account?: Account | Address;
+    collection: Address;
+  }) {
+    const signatureResponse = (await this.get(
+      `${ZORA_API_BASE}signature/${
+        this.network.zoraBackendChainName
+      }/${collection.toLowerCase()}/${uid}`,
+    )) as PremintResponse;
+
+    const signerData = {
+      ...signatureResponse,
+      premint: {
+        ...convertPremint(signatureResponse.premint),
+        deleted: true,
+      },
+    };
+
+    return await this.signAndSubmitPremint({
+      walletClient,
+      account,
+      checkSignature: false,
+      verifyingContract: collection,
+      publicClient: this.getPublicClient(),
+      uid: uid,
+      collection: signerData.collection,
+      premintConfig: signerData.premint,
+    });
+  }
+
+  private async signAndSubmitPremint({
+    walletClient,
+    publicClient,
+    verifyingContract,
+    premintConfig,
+    uid,
+    account,
+    checkSignature,
+    collection,
+  }: {
+    publicClient: PublicClient;
+    uid: number;
+    walletClient: WalletClient;
+    verifyingContract: Address;
+    checkSignature: boolean;
+    account?: Address | Account;
+    premintConfig: PremintConfig;
+    collection: PremintResponse["collection"];
+  }) {
+    if (!account) {
+      account = walletClient.account;
+    }
+    if (!account) {
+      throw new Error("No account provided");
+    }
+
+    const signature = await walletClient.signTypedData({
+      account,
+      ...preminterTypedDataDefinition({
+        verifyingContract,
+        premintConfig,
+        chainId: this.chain.id,
+      }),
+    });
+
+    if (checkSignature) {
+      const [isValidSignature] = await publicClient.readContract({
+        abi: zoraCreator1155PremintExecutorImplABI,
+        address: this.getExecutorAddress(),
+        functionName: "isValidSignature",
+        args: [collection, premintConfig, signature],
+      });
+      if (!isValidSignature) {
+        throw new Error("Invalid signature");
+      }
+    }
+
+    const apiData = {
+      collection,
+      premint: encodePremintForAPI(premintConfig),
+      chain_name: this.network.zoraBackendChainName,
+      signature: signature,
+    };
+
+    let premint = await this.retries(() =>
+      this.post(`${ZORA_API_BASE}signature`, apiData),
+    );
+
+    return {
+      zoraUrl: `https://${
+        this.network.isTestnet ? "testnet." : ""
+      }zora.co/collect/${
+        this.network.zoraPathChainName
+      }:${verifyingContract}/premint-${uid}`,
+      uid,
+      verifyingContract,
+      premint,
+    };
+  }
+
+  private async retries<T>(
```

Comment:
this should be extracted to a static private function since it doesn't rely on `this` - id try to minimize the size of this class to make it easier to scan and be more cohesive